sound/soc/soc-pcm.c

```diff
@@ -2084,9 +2111,17 @@ int dpcm_be_dai_trigger(struct snd_soc_pcm_runtime *fe, int stream,
 				spin_unlock_irqrestore(&fe->card->dpcm_lock, flags);
 				continue;
 			}
+			if (be->dpcm[stream].state == SND_SOC_DPCM_STATE_START)
+				be->dpcm[stream].be_start--;
+
+			if ((be->dpcm[stream].state == SND_SOC_DPCM_STATE_START &&
+			     be->dpcm[stream].be_start == 0) ||
+			    be->dpcm[stream].state == SND_SOC_DPCM_STATE_PAUSED)
+				do_trigger = true;
```

Comment:
is this check correct @plbossart ? irrespective of whether the current state is START or PAUSED, we should set do_trigger to true is be_start == 0 no?

Reply:
I think you're right. I'll update this, thanks for the suggestion.

Reply:
done